types/index.d.ts

```diff
@@ -774,6 +807,40 @@ declare module 'cloudinary' {
             function usage(callback?: ResponseCallback, options?: AdminApiOptions): Promise<any>;
 
             function usage(options?: AdminApiOptions): Promise<any>;
+
+            /****************************** Structured Metadata API V2 Methods *************************************/
+
+            function add_metadata_field(field: MetadataField, options?: AdminApiOptions, callback?: ResponseCallback): Promise<MetadataField>;
+
+            function add_metadata_field(field: MetadataField, callback?: ResponseCallback): Promise<MetadataField>;
+
+            function list_metadata_fields(callback?: ResponseCallback, options?: AdminApiOptions): Promise<MetadataFields>;
+
+            function list_metadata_fields(options?: AdminApiOptions): Promise<MetadataFields>;
```

Comment:
I think there's a difference between the response of MetadataFields and the argument of MetadataFields

The argument MetadataFields can have missing parts, can the response also have missing parts?
For example, if MetadataField.label is optional, when you fetch it from the server, is will that key be missing?

Reply:
Created two separate interfaces. one for the response and one for the options.